pkg/metrics/channels.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type ChannelMetrics[T any] map[string]<-chan T
```

Comment:
nit: use a slice instead of map since we only iterate over the channels

Reply:
Dictionary is for the names. Otherwise I have to hardcode them.